setup.py

```diff
@@ -16,13 +16,39 @@
 import subprocess
 import shutil
 import sys
-import toml
+
+from importlib import import_module
+from importlib.util import find_spec
 
 from pathlib import Path
 from setuptools import setup, Extension, find_namespace_packages
 from setuptools.command.build_ext import build_ext
 
-project_name = toml.load("pyproject.toml")['project']['name']
+
+has_toml = False
+toml_libs = ["tomli", "tomllib", "tomlkit", "toml"]
+for pkg in toml_libs:
+    spec = find_spec(pkg)
+    if spec:
```

Comment:
```suggestion
    if find_spec(pkg):
```

Reply:
Let me keep this `find_spec` call before if-cond as it'd help debugging the logic specially when `find_spec` fails and you'd like to lookup/traverse these lines in the call stack.